api/src/main/java/jakarta/data/page/Slice.java

```diff
@@ -0,0 +1,85 @@
+package jakarta.data.page;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+public interface Slice<T> extends Iterable<T> {
```

Comment:
Please let's not bring back `Slice`. What do we need it for?

Reply:
I think Otavio is proposing to get rid of page numbers entirely with cursor-based pagination.  I think that would be a mistake.  While some users of cursor-based pagination will not want or care about page numbers, other users of cursor-based pagination will.  The spec is currently very lazily written with respect to page numbers when using cursor-based pagination, which is my fault, and says not to rely on them.  Given recent discussion, it sounds like we will need to add more guarantees in the spec to ensure they can be useful when they are wanted.  Getting rid of them completely doesn't seem like a good idea.

Reply:
> I think Otavio is proposing to get rid of page numbers entirely with cursor-based pagination. I think that would be a mistake. 

Yeah I guess it's not what I would have done.

But if we have page numbers, they have to be consistent, at least in the absence of phantom reads.

Reply:
I need a common interface for both OffSetPage and CursoredPage.

I will use Page as the main interface instead of Slice.

Reply:
@gavinking @njr-11 I returned the method, and I also renamed it to OffSetPage